spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/integration/outbound/PubSubMessageHandler.java

```diff
@@ -89,16 +88,18 @@ protected void handleMessageInternal(Message<?> message) throws Exception {
 
 		ByteString pubsubPayload;
 
-		if (payload instanceof byte[]) {
+		if (this.messageConverter != null) {
+			pubsubPayload = ByteString.copyFrom((byte[]) this.messageConverter.fromMessage(message,
+					byte[].class));
+		}
+		else if (payload instanceof byte[]) {
 			pubsubPayload = ByteString.copyFrom((byte[]) payload);
 		}
 		else if (payload instanceof ByteString) {
 			pubsubPayload = (ByteString) payload;
 		}
 		else {
-			pubsubPayload =	ByteString.copyFrom(
-					(String) this.messageConverter.fromMessage(message, String.class),
-					Charset.defaultCharset());
+			pubsubPayload = ByteString.copyFrom(message.toString().getBytes());
```

Comment:
This `message.toString()` seems to just give the name of the class and the hashcode, and not the actual contents. Why not keep the StringMessageConverter that we had previously and use that to provide some semblance of the contents?